src/backends/interflop-vprec/interflop_vprec.c

```diff
@@ -455,6 +457,96 @@ static inline double _vprec_binary64_binary_op(double a, double b,
   return res;
 }
 
+// Round the float with the given precision
+static float _vprec_round_binary32( float a,
+                                  int binary32_range,
+                                  int binary32_precision)
+{
+  if (!isfinite(a)) {
+    return a;
+  }
+
+  /* round to zero or set to infinity if underflow or overflow compare to
+   * VPRECLIB_BINARY32_RANGE */
+  int emax = (1 << (binary32_range - 1)) - 1;
+  int emin = (emax > 1) ? 1 - emax : -1;
+
+  binary32 aexp = {.f32 = a};
+
+  aexp.s32 = ((FLOAT_GET_EXP & aexp.u32) >> FLOAT_PMAN_SIZE) - FLOAT_EXP_COMP;
+
+  /* check for overflow or underflow in target range */
+  bool sp_case = false;
+  if (aexp.s32 > emax) {
+    a = a * INFINITY;
+    sp_case = true;
+  }
+
+  if (aexp.s32 <= emin) {
+    a = handle_binary32_denormal(a, emin, aexp.u32, binary32_precision);
+  }
+
+  /* Specials ops must be placed after denormal handling  */
+  /* If one of the operand raises an underflow, the operation */
+  /* has a different behavior. Example: x*Inf != 0*Inf */
+
+  if (sp_case) {
+    return a;
+  }
+
+  /* else, normal case: can be executed even if a
+     previously rounded and truncated as denormal */
+  if (binary32_precision < FLOAT_PMAN_SIZE) {
+    a = round_binary32_normal(a, binary32_precision);
+  }
+
+  return a;
+}
+
+// Round the double with the given precision
+static double _vprec_round_binary64(double a, 
```

Comment:
There seems to be a lot in common with _vprec_binary64_binary_op.
There should be some factoring here.

Reply:
_vprec_round_binaryXX functions are now used in _vprec_binaryXX_binary_op to round inputs and outputs